### ISSUE TITLE
strange null pointer exception in 3.14.0 version - maybe a race condition?

### ISSUE THREAD

**Issue by pedrofurtado** [OneSignal/OneSignal-Android-SDK#1040]
Stacktrace:

```
java.lang.NullPointerException: 
  at com.onesignal.OneSignalCacheCleaner$2.run (OneSignalCacheCleaner.java:116)
  at java.lang.Thread.run (Thread.java:818)
```

![Capturar](https://user-images.githubusercontent.com/10530520/83471719-cc03c600-a45b-11ea-82b6-91e2737065e9.PNG)

**Comment by ngbaptista** [comment]
I have a similar error

<img width="966" alt="screencapture 2020-06-03 at 09 13 10" src="https://user-images.githubusercontent.com/66356980/83612679-a1f4f580-a57a-11ea-8968-d33f8ac8ab5d.png">

**Comment by rgomezp** [comment]
Howdy,
We are aware of this new issue. Thanks for your patience while we resolve it. 

In the meantime, can you please provide all the information we request in the issue template?

Cheers

**Comment by ngbaptista** [comment]
Description:

null pointer exception on 3.14.0 version

Environment:

Galaxy Tab S2; android version 7.0; free RAM memory 782.45 MB; free disk space 8.01 GB
Galaxy S7; android version 8.0.0;  free RAM memory 1.84 GB; free disk space 10.08 GB
Galaxy S6 edge; android version 7.0;  free RAM memory 1015.19 MB; free disk space 17.84 GB
ONEPLUS A3003: android version 9;  free RAM memory 2.55 GB; free disk space 8.85 GB

Steps to Reproduce Issue:
these are errors on production environment, happened to end users, we don't have the steps.
Anything else:

Fatal Exception: java.lang.NullPointerException: Attempt to invoke interface method 'boolean android.database.Cursor.isClosed()' on a null object reference
       at com.onesignal.OneSignalCacheCleaner$2.run(OneSignalCacheCleaner.java:116)
       at java.lang.Thread.run(Thread.java:764)

**Comment by rgomezp** [comment]
Fixed by #1047

**Comment by pedrofurtado** [comment]
In which version will be available? Is there some ETA?
